transformer_engine/jax/attention.py

```
@@ -68,7 +71,9 @@ def is_bottom_right(self):
         return self in [
             AttnMaskType.CAUSAL_BOTTOM_RIGHT_MASK,
             AttnMaskType.PADDING_CAUSAL_BOTTOM_RIGHT_MASK,
-        ]
+        ] or ( #Use bottom-right if no mask is set to match TE FusedAttn (CK) code
+            is_hip_extension() and self == AttnMaskType.NO_MASK
```

I thought top-left or bottom-right is only valid for the scenarios where there is a mask (causal or sliding window). Does top-left or bottom-right mean when there is no mask?

---

This method is called from https://github.com/ROCm/TransformerEngine/pull/178/commits/2f85f5f23f3fcdf908b595854a6dd80fa4af1236#diff-d6e9c5eeb7a2c944a7123af74042744a9b8dd78234b40942ce76ae71d45ad02aR154 to determine which mask to generate for SWA. So the change means that default mask (when no type is specified) will be bottom-right.